lib/sinon/collection.js

```diff
@@ -46,7 +46,13 @@ var collection = {
     },
 
     resetHistory: function resetHistory() {
-        each(this, "resetHistory");
+        getFakes(this).forEach(function (fake) {
+            var method = fake.resetHistory || fake.reset;
+
+            if (method) {
```

Comment:
> If you're going for the check, shouldn't it throw if there is no way to reset the thing this is called on?

I didn't want to create something that was vastly different from what `each` does in the same file.

https://github.com/sinonjs/sinon/blob/master/lib/sinon/collection.js#L19-L28